src/vma/dev/time_converter.cpp

```diff
@@ -156,11 +156,14 @@ ts_conversion_mode_t time_converter::update_device_converters_status(net_device_
 
 void time_converter::clean_obj()
 {
-	set_cleaned();
+	if (is_cleaned()) {
```

Comment:
Can't we better reuse the code by putting the clean logic inside the base class?

Reply:
Probably it is possible if all classes inherited from cleanable_obj must registrate timers. In current code in some case we use locking using inherited class lock.